includes/class-wc-invoice-payment-rest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+use Dompdf\Dompdf;
+
+/**
+ * @see       https://www.linknacional.com/
+ * @since      1.2.0
+ */
+final class Wc_Payment_Invoice_Loader_Rest {
+    public function register_routes(): void {
+        register_rest_route(
+            'wc-invoice-payment/v1',
+            '/generate-pdf',
+            array(
+                'methods' => 'GET',
+                'callback' => array($this, 'generate_invoice'),
+                'permission_callback' => array($this, 'check_permission'),
+            )
+        );
+    }
+
+    public function check_permission() {
+        return current_user_can('administrator');
+    }
+
+    public function generate_invoice(WP_REST_Request $request): void {
+        $invoice_id = $request->get_param( 'invoice_id' );
+
+        $getHtml = function () use ($invoice_id) {
+            $order = wc_get_order($invoice_id);
+
+            $invoice_pdf_template_id = $order->get_meta('wcip_select_invoice_template_id');
+
+            if (empty($invoice_pdf_template_id) || 'global' === $invoice_pdf_template_id) {
+                $invoice_pdf_template_id = get_option('lkn_wcip_global_pdf_template_id', 'linknacional');
+            }
+
+            $template_file_path = __DIR__ . "/templates/$invoice_pdf_template_id/main.php";
+
+            if ( ! file_exists($template_file_path)) {
+                return require_once __DIR__ . "/templates/linknacional/main.php";
+            }
+
+            return require_once $template_file_path;
+        };
+        // Prints the HTML
```

Comment:
@brunoferreiralkn Talvez remover esses comentários.

Reply:
These comments are useful for debugging while in development!